kaitaistruct.py

```diff
@@ -279,57 +351,67 @@ def read_bits_int_le(self, n):
     # Byte arrays
     # ========================================================================
 
-    def read_bytes(self, n):
+    def alignment(self, a):
+        return (a - self.pos()) % a
+
+    def read_bytes(self, n, align=0):
         if n < 0:
-            raise ValueError(
-                "requested invalid %d amount of bytes" %
-                (n,)
-            )
+            raise ValueError("%d is invalid amount of bytes" % n)
         r = self._io.read(n)
         if len(r) < n:
-            raise EOFError(
-                "requested %d bytes, but got only %d bytes" %
-                (n, len(r))
-            )
+            raise EOFError("got only %d bytes out of %d requested" % (n, len(r)))
+        if align > 1:
+            self._io.seek(self.alignment(align), 1)
         return r
 
+    def write_bytes(self, data, align=0, pad=0, padding=b'\0'):
+        if data is None:
+            return
+        nb = len(data)
+        if nb == 0 and align < 2 and pad < 1:
+            return
+        if self._io.write(data) != nb:
+            raise Exception("not all bytes written")
+        if pad > 0:
+            self._io.write(padding * pad)
+        if align > 1:
+            self._io.write(padding * self.alignment(align))
+        return
+
     def read_bytes_full(self):
         return self._io.read()
 
-    def read_bytes_term(self, term, include_term, consume_term, eos_error):
+    def read_bytes_term(self, term, include_term=False, consume_term=True, eos_error=True, elem_size=1):
```

Comment:
As above, what's the purpose of the `elem_size` parameter, and how exactly is it related to serialization?

Also, I'm not sure if it's very useful to add default values for existing arguments. All code that uses this API is generated by the compiler, which already sets all these arguments explicitly, so these defaults would never be used.

Reply:
Users can change code generated by compiller because it can't cover all use cases, in this way the code may be called manually. It's a chore to list all parameters in this case.

Reply:
For example `elem_size` can be used to deal with 2 or 4 byte strings.